apps/trigger-oracle/src/lib.rs

```diff
@@ -285,17 +285,19 @@ impl TriggerExecutor for OracleTrigger {
         );
         loops.push(manager);
 
-        let (tr, _, rest) = futures::future::select_all(loops).await;
+        loop {
+            let (tr, _, rest) = futures::future::select_all(loops).await;
 
-        drop(rest);
-        match tr {
-            Ok(TerminationReason::ExitRequested) => {
-                tracing::trace!("Exiting");
-                Ok(())
-            }
-            _ => {
-                tracing::error!("Fatal: {:?}", tr);
-                Err(anyhow::anyhow!("{tr:?}"))
+            match tr {
+                Ok(TerminationReason::ExitRequested) => {
+                    tracing::trace!("Exiting");
+                    drop(rest);
+                    return Ok(());
+                }
+                _ => {
```

Comment:
Can you please add new `TerminationReason::OracleExecutionFailed` or something like this and continue waiting for the other futures if this happens.
If any other loop crashes we better stop the whole reporter, or try to restart it for sequencer or orchestrator. Otherwise I like the pattern.

Reply:
Sounds good, on it.

Reply:
Done.